src/toxcore/dht/server.rs

```diff
@@ -723,17 +755,15 @@ mod tests {
         // error case, empty kbucket
         let nrq = NodesRequestPayload { pk: bob_pk, id: random_u64() };
         let nodes_req = NodesRequest::new(&precomp, &bob_pk, nrq);
-        let client = alice.create_client(&addr, bob_pk);
-        assert!(alice.handle_nodes_req(client, nodes_req).wait().is_err());
+        assert!(alice.handle_nodes_req(nodes_req, addr).wait().is_err());
```

Comment:
You should test it by passing special packet and calling `handle_packet`, not by calling handle_YOUR_PACKET directly. See https://github.com/tox-rs/tox/blob/master/src/toxcore/tcp/server/server.rs#L430